### PR TITLE
feat(tencentcloud): init tencentcloud

### PR DESCRIPTION
**Pull request by zexiplus**
This pr is the first commit for the tencentcloud provider and includes the intialization works. The cloud provider proposal is [spinnaker/governance#79](https://github.com/spinnaker/governance/issues/79)

### CODE REVIEW

**Comment by zexiplus** [comment]
would you like to take a moment to review this PR, I 'd preciate to take you advice, thx @erikmunson

File: app/scripts/modules/tencentcloud/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@spinnaker/tencentcloud",
```
Comment (caseyhebebrand) [inline app/scripts/modules/tencentcloud/package.json:2]: Don't forget to include the version in this file.

Comment (zexiplus) [reply app/scripts/modules/tencentcloud/package.json:2]: > Don't forget to include the package version in this file.

Thanks for your guide. 
But if I add version field in the package.json, I will not pass the actions, cuz the previous version is empty, it's not equal to 0.0.1, how can I fix it ? Do I need to rewrite assert_package_bumps_standalone.sh or ... ?
image
https://github.com/spinnaker/deck/runs/614469005?check_suite_focus=true

Comment (zexiplus) [reply app/scripts/modules/tencentcloud/package.json:2]: > Don't forget to include the package version in this file.

Thanks for your guide. 
But if I add version field in the package.json, I will not pass the actions, cuz the changes in package.json is not only version field but also includes the other fields, in addition, the previous version is empty. It cause that the version change is not only change in package.json and fail the action's pipeline. How could I fix it ? Do I need to rewrite assert_package_bumps_standalone.sh or ... ?
image
https://github.com/spinnaker/deck/runs/614469005?check_suite_focus=true

Comment (caseyhebebrand) [reply app/scripts/modules/tencentcloud/package.json:2]: You can always merge this PR with version `0.0.0` . Then if the action fails, you can follow up with a simple PR that bumps the version to `0.0.1` so the action will be triggered.

Comment (zexiplus) [reply app/scripts/modules/tencentcloud/package.json:2]: > You can always merge this PR with version `0.0.0` . Then if the action fails, you can follow up with a simple PR that bumps the version to `0.0.1` so the action will be triggered.

Thanks for your suggestion, I added version `0.0.0` to package.json

Comment (caseyhebebrand) [reply app/scripts/modules/tencentcloud/package.json:2]: If you keep running into build failures with this, you can also divide this into 2 PRs. The first would include all the module set up and the second would add the config for the workflow.

Comment (zexiplus) [reply app/scripts/modules/tencentcloud/package.json:2]: > If you keep running into build failures with this, you can also divide this into 2 PRs. The first would include all the module set up and the second would add the config for the workflow.

As the first PR,  this time I remove the tencentcloud config for the workflow and version field in package.json. I will add the version field and config for the workflow in the next PR. Is it all right?  If not could you give me some suggestions with more detail?

**Comment by zexiplus** [comment]
@christopherthielen,  would you like to take a moment to review this PR, I 'd preciate to take you advice, thx
additional remarks: 
I did not add version in tencentcloud/package.json cuz I want to pass the action.  If this PR can be merged into master, I will add it in the next PR.

File: app/scripts/modules/tencentcloud/src/tencentcloud.setting.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    vpc?: string;
+  };
+  defaultSecurityGroups?: string[];
+  loadBalancers?: {
+    inferInternalFlagFromSubnet: boolean;
+    certificateTypes?: string[];
+  };
+  useAmiBlockDeviceMappings?: boolean;
+  classicLaunchLockout?: number;
+  classicLaunchWhitelist?: IClassicLaunchWhitelist[];
```
Comment (christopherthielen) [inline app/scripts/modules/tencentcloud/src/tencentcloud.setting.ts:22]: Does tencent really have AmiBlockDeviceMappings or ClassicLaunch?  Both of these things imply Amazon to me.

File: halconfig/settings.js [hunk 3]
```diff
@@ -113,6 +113,12 @@ var oracle = {
     region: '{%oracle.default.region%}',
   },
 };
+var tencentcloud = {
+  defaults: {
+    account: '{%oracle.default.account%}',
```
Comment (christopherthielen) [inline halconfig/settings.js:118]: `oracle`?

Comment (zexiplus) [reply halconfig/settings.js:118]: sorry, that's was a mistake

Comment (zexiplus) [reply halconfig/settings.js:118]: > `oracle`?

sorry, that's was a mistake

File: app/scripts/modules/tencentcloud/src/tencentcloud.setting.ts (hunk 2 above)
Comment (zexiplus) [reply app/scripts/modules/tencentcloud/src/tencentcloud.setting.ts]: > I found a couple references to other cloud providers that look possibly misplaced.

Thanks for your earnest review, I fixed them

**Comment by christopherthielen** [comment]
@zexiplus once this branch is up to date with `master` it will get merged.  We want PRs to have "allow updates" enabled so mergify.io can do its automatic merging.

See: https://github.com/spinnaker/deck/pull/8210/checks?check_run_id=673181817

**Comment by zexiplus** [comment]
> @zexiplus once this branch is up to date with `master` it will get merged. We want PRs to have "allow updates" enabled so mergify.io bot can do its automatic merging.
> 
> See: https://github.com/spinnaker/deck/pull/8210/checks?check_run_id=673181817

Thx, already updated and merged

**Pull request merged**